src/mpid/ch4/src/ch4_impl.h

```diff
@@ -15,10 +15,13 @@
 #include <mpidch4.h>
 #include "mpidig.h"
 
-MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_test(int flags);
+MPL_STATIC_INLINE_PREFIX int MPIDI_Progress_test(int hook_flags, int global_progress, int vci_is_nm,
+                                                 int vci);
```

Comment:
`vci_is_nm` seems weird.  I'd recommend `vci_type` which is an enum (NM vs. SHM).

Reply:
I agree, I will address this.

Reply:
Can you use an `enum` instead of `int` and use readable names like `MPIDI_VCI_TYPE__NM`, instead of `1` and `0`?

Similarly, `global_progress`: maybe that can be an enum with names `MPIDI_PROGRESS_TYPE__GLOBAL` and `MPIDI_PROGRESS_TYPE__VCI`.